tests/bluetooth/bsim_bt/bsim_test_l2cap_stress/src/common.c

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "common.h"
+
+#define LOG_MODULE_NAME common
+
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(LOG_MODULE_NAME, LOG_LEVEL_DBG);
+
+extern enum bst_result_t bst_result;
+
+void test_init(void)
+{
+	bst_ticker_set_next_tick_absolute(WAIT_TIME);
+	bst_result = In_progress;
+}
+
+void test_tick(bs_time_t HW_device_time)
+{
+	if (bst_result != Passed) {
+		FAIL("test failed (not passed after %i seconds)\n", WAIT_SECONDS);
+	}
+}
+
+/* Call in init functions*/
+void device_sync_init(uint device_nbr)
+{
+	uint peer = CENTRAL_ID;
+
+	if (device_nbr == CENTRAL_ID) {
+		peer = PERIPHERAL_ID;
+	}
+
+	uint dev_nbrs[BACK_CHANNELS] = {peer};
+	uint channel_nbrs[BACK_CHANNELS] = {0};
+	uint *ch = bs_open_back_channel(device_nbr, dev_nbrs, channel_nbrs, BACK_CHANNELS);
```

Comment:
Shouldn't `BACK_CHANNELS` just be `ARRAY_SIZE(channel_nbrs)`?

Reply:
no idea, originally this was the l2cap ecred test that I hacked until I stressed the system enough for the bug to manifest itself. I think I can do away with the backchannels entirely, just relying on stack callbacks. As i said in another comment, the code is still very hacky, so don't think it makes sense to review the test part right now.